ios/Classes/FlutterDocumentView.m

```diff
@@ -113,7 +137,158 @@ - (void)openDocument:(NSString *)document password:(NSString *)password config:(
         fileURL = [NSURL fileURLWithPath:document];
     }
     
+    _documentViewController.delegate = self;
+    flutterResult = result;
+    
     [_documentViewController openDocumentWithURL:fileURL password:password];
 }
 
+- (void)importAnnotationCommand:(NSString *)xfdfCommand
+{
+    if(_documentViewController.document == Nil )
+    {
+        // something is wrong, no document.
+        NSLog(@"Error: The document view controller has no document.");
+        return;
+    }
+    
+    NSError* error;
+    
+    [_documentViewController.pdfViewCtrl DocLock:YES withBlock:^(PTPDFDoc * _Nullable doc) {
+        if( [doc HasDownloader] )
+        {
+            // too soon
+            NSLog(@"Error: The document is still being downloaded.");
+            return;
+        }
+
+        PTFDFDoc* fdfDoc = [doc FDFExtract:e_ptboth];
+        [fdfDoc MergeAnnots:xfdfCommand permitted_user:@""];
+        [doc FDFUpdate:fdfDoc];
+
+        [_documentViewController.pdfViewCtrl Update:YES];
+
+
+    } error:&error];
```

Comment:
You can check the result of this error object and emit a log statement as is done in the method `-(void)saveDocument:(FlutterResult)result`.